nipy/modalities/fmri/design.py

```diff
@@ -421,3 +422,104 @@ def stack_designs(*pairs):
+
+    Parameters
+    ----------
+    ons_dur_amp : str or array
+        Path to OpenFMRI stimulus file or 2D array containing three columns
+        corresponding to onset, duration, amplitude.
+
+    Returns
+    -------
+    block_spec : array
+        Structured array with fields "start" (corresponding to onset time),
+        "end" (onset time plus duration), "amplitude".
+    """
+    if not isinstance(ons_dur_amp, np.ndarray):
+        ons_dur_amp = np.loadtxt(ons_dur_amp)
+    onsets, durations, amplitudes = ons_dur_amp.T
+    return make_recarray(
+        np.column_stack((onsets, onsets + durations, amplitudes)),
+        names=['start', 'end', 'amplitude'],
+        drop_name_dim=True)
+
+
+def block_amplitudes(name, block_spec, t, hrfs=(glover,),
+                     convolution_padding=5.,
+                     convolution_dt=0.02,
+                     hrf_interval=(0.,30.)):
+    """ Design matrix at times `t` for blocks specification `block_spec`
+
+    Create design matrix for linear model from a block specification
+    `block_spec`,  evaluating design rows at a sequence of time values `t`.
+
+    `block_spec` may specify amplitude of response for each event, if different
+    (see description of `block_spec` parameter below).
+
+    The on-off step function implied by `block_spec` will be convolved with
+    each HRF in `hrfs` to form a design matrix shape ``(len(t), len(hrfs))``.
+
+    Parameters
+    ----------
+    name : str
+        Name of condition
+    block_spec : np.recarray or array-like
+       A recarray having fields ``start, end, amplitude``, or a 2D ndarray /
+       array-like with three columns corresponding to start, end, amplitude.
+    t : np.ndarray
+       An array of np.float values at which to evaluate the design. Common
+       examples would be the acquisition times of an fMRI image.
+    hrfs : sequence, optional
+       A sequence of (symbolic) HRFs that will be convolved with each block.
+       Default is ``(glover,)``.
+    convolution_padding : float, optional
+       A padding for the convolution with the HRF. The intervals
+       used for the convolution are the smallest 'start' minus this
+       padding to the largest 'end' plus this padding.
+    convolution_dt : float, optional
+       Time step for high-resolution time course for use in convolving the
+       blocks with each HRF.
+    hrf_interval: length 2 sequence of floats, optional
+       Interval over which the HRF is assumed supported, used in the
+       convolution.
+
+    Returns
+    -------
+    X : np.ndarray
+       The design matrix with ``X.shape[0] == t.shape[0]``. The number of
+       columns will be ``len(hrfs)``.
+    contrasts : dict
+       A contrast is generated for each HRF specified in `hrfs`.
+    """
+    block_spec = np.asarray(block_spec)
+    if block_spec.dtype.names is not None:
+        if not block_spec.dtype.names in (('start', 'end'),
```

Comment:
very minor -- IIRC it is recommended to use "x not in",  not "not x in"  (primarily for readability iirc)